### DISCUSSION TITLE
Please 🙏 clean up stabilityDays related discussions, issues, docs.

### DISCUSSION THREAD

**Discussion by TWiStErRob** [renovatebot/renovate#22520] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Wanted end result.

Best persons to look at this: @rarkins @setchy

(NOTE: I deliberately did not use any links and issue references to confuse things further, please use the numbers to open the issues. I added their title for easier reading.)

Updates to some issues, depicting current real state. Here's my proposal:
 * Update issue `12914 Improved age filtering/controls`
   - [x] `rarkins` Edit OP replacing "`release age (stabilityDays.. needs renaming)`" with a link to the issue/PR that solved this. This will show that progress has been made.
   - [x] `rarkins` Edit OP removing "`Maybe rename stabilityDays`"
   - [x] `rarkins` Link to 21442 as related
 - [x] `setchy` Rename issue `21013 Improve Stability Days documentation and link to from stabilityDays builds`
 - [x] `rarkins` Close discussion `14948 Point to stabilityDays docs instead of homepage for our renovate/stabilitydays CI check` and link to tracking issue `21013` instead. (Right now it looks like two discussions just point to each other.)
 - [x] `setchy` Accept answer https://github.com/renovatebot/renovate/discussions/21008#discussioncomment-5347127 to strengthen the link to the tracking issue `21013` instead of referencing an unanswered open discussion.
 - [x] `anyone` Add a the `stabilityDays` "word" in the documentation of [minimumReleaseAge](https://docs.renovatebot.com/configuration-options/#minimumreleaseage) so that Google and direct searches based on existing blog posts out there find the right information. Happy to do this myself if you agree.

### What you tried so far.

It all started by wanting to delay PR creation, I remembered `stabilityDays` was a thing in Renovate, so I looked. I was looking in the docs:
![image](https://github.com/renovatebot/renovate/assets/2906988/ee372ffc-13e2-4317-8bf4-8dd735f55090)

This is where the real head-scratching began... I started looking in Google and it found random discussions and multiple places linking to [this dead page](https://docs.renovatebot.com/configuration-options/#stabilitydays), so it confirmed this thing existed.

Multiple discussions also linked to an open issue and all info suggested this **might** be renamed in the future. Since the issue is still open and the wording refers to the future, it just confused me more.

After digging in GitHub issue search I found the PR that did the rename, but there was no real link to/from those existing issues.

After this I was curious what's missing, why was it so hard to find the info and mapped the whole thing:

```mermaid
graph TB;
    12914[open feature\nissue#12914\nImproved age filtering/controls]
    style 12914 fill:#1f883d,color:white
    14948[unanswered\ndiscussions#14948\nPoint to stabilityDays docs instead of homepage for our renovate/stabilitydays CI check]
    style 14948 fill:white,color:#656d76
    16875[closed\nissue#16875\nminimumReleaseAge instead of stabilityDays]
    style 16875 fill:#8250df,color:white
    21008[closed unanswered\ndiscussion#21008\nBitbucket Cloud: Should the renovate/stability-days build hyperlink to the stabilityDays configuration option?]
    style 21008 fill:#8250df,color:1f883d
    21013[open\nissue#21013\nImprove Stability Days documentation and link to from stabilityDays builds]
    style 21013 fill:#1f883d,color:white
    21376[merged\npull#21376\nfeat/config: rename stabilityDays to minimumReleaseAge]
    style 21376 fill:#8250df,color:white
    21442[open\nissue#21442\nChange the renovate/stability-days status check to renovate/minimum-release-age]
    style 21442 fill:#1f883d,color:white

    14948 -- Warning: We'll need to update the link if we rename stabilityDays to something else like minimumReleaseAge --> 12914
    14948 -- jinx great minds think alike --> 21013
    16875 -- discussed some new names for the stabilityDays config option --> 12914
    16875 -- Fixed by --> 21376
    21008 -- Closing discussion in favor of this more rich question --> 14948
    21008 -- raised as --> 21013
    21013 -- keep in mind we would like to rename stabilityDays --> 12914
    21013 -- more thoroughly documented discussion --> 14948
    21013 -- Based off this discussion --> 21008
    21376 -- Closes --> 16875
    21376 -- I couldn't find the issue yet, so I created one --> 21442
```

Based on this I propose some quick admin to help users find things easier.

### Relevant debug logs

None

**Comment by HonkingGoose** [comment]
We _really_ had a config option called `stabilityDays`, we renamed it to `minimumReleaseAge` with this PR:

- https://github.com/renovatebot/renovate/pull/21376

We did not clean up the references to `stabilityDays` in our repository after the merge. I would do it now, but I don't have the rights to edit issue titles or issue comments. So this is cleanup for one of the maintainers. 😄

**Comment by HonkingGoose** [comment]
> `anyone` Add a the `stabilityDays` "word" in the documentation of [minimumReleaseAge](https://docs.renovatebot.com/configuration-options/#minimumreleaseage) so that Google and direct searches based on existing blog posts out there find the right information. Happy to do this myself if you agree.

@TWiStErRob I don't like this idea, I think the docs should only have _current_ terms. The end-goal for Renovate is to raise config migration PRs automatically, to update users from old terms to new terms:

- #16359

Once we have that, the users configuration will always be in sync with the Renovate docs. At least if the users actually update to the latest version of Renovate in good time... 😄

**Reply by TWiStErRob** [reply]
That north star sounds awesome, but the issue still exists:

**Who will migrate blog posts, cheat-sheets, company internal documentations, tutorials, etc?**
 * https://www.mend.io/resources/blog/renovating-with-renovate/
 * https://www.augmentedmind.de/2021/07/25/renovate-bot-cheat-sheet/
 * https://bahr.dev/2022/12/05/automatic-depdency-upgrades/

If I read one of these and want to use `stabilityDays`, because it's useful for me, yet I can't find it in the documentation, then what do I do? Just add it blindly and hope that it works? Notice that these are not even that old, and one of them is "first party"!

I really think the only feasible long term solution is to have some way of expressing "this config entry had these a, b, c names in the past" or "this config entry was split/merged from a, b, c". What I agree about is that maybe this is discussion is a bit more focused than this general idea. :)

**Reply by TWiStErRob** [reply]
Also https://github.com/renovatebot/renovate/issues/16359#issuecomment-1570530118

**Reply by HonkingGoose** [reply]
@TWiStErRob by the way, thank you for gathering all the information and making a cool graph! ❤️

**Reply by TWiStErRob** [reply]
<details><summary>Mermaid ftw!</summary>

````
```mermaid
graph TB;
    12914[open feature\nissue#12914\nImproved age filtering/controls]
    style 12914 fill:#1f883d,color:white
    14948[unanswered\ndiscussions#14948\nPoint to stabilityDays docs instead of homepage for our renovate/stabilitydays CI check]
    style 14948 fill:white,color:#656d76
    16875[closed\nissue#16875\nminimumReleaseAge instead of stabilityDays]
    style 16875 fill:#8250df,color:white
    21008[closed unanswered\ndiscussion#21008\nBitbucket Cloud: Should the renovate/stability-days build hyperlink to the stabilityDays configuration option?]
    style 21008 fill:#8250df,color:1f883d
    21013[open\nissue#21013\nImprove Stability Days documentation and link to from stabilityDays builds]
    style 21013 fill:#1f883d,color:white
    21376[merged\npull#21376\nfeat/config: rename stabilityDays to minimumReleaseAge]
    style 21376 fill:#8250df,color:white
    21442[open\nissue#21442\nChange the renovate/stability-days status check to renovate/minimum-release-age]
    style 21442 fill:#1f883d,color:white

    14948 -- Warning: We'll need to update the link if we rename stabilityDays to something else like minimumReleaseAge --> 12914
    14948 -- jinx great minds think alike --> 21013
    16875 -- discussed some new names for the stabilityDays config option --> 12914
    16875 -- Fixed by --> 21376
    21008 -- Closing discussion in favor of this more rich question --> 14948
    21008 -- raised as --> 21013
    21013 -- keep in mind we would like to rename stabilityDays --> 12914
    21013 -- more thoroughly documented discussion --> 14948
    21013 -- Based off this discussion --> 21008
    21376 -- Closes --> 16875
    21376 -- I couldn't find the issue yet, so I created one --> 21442
```
````

</summary>

**Comment by rarkins** [comment, marked as answer]
I don't think a full clean up is necessary if the new option had a note saying "Formerly known as stabilityDays" which would be found when searching the docs. Open issues naturally should have renaming done

**Reply by HonkingGoose** [reply]
> I don't think a full clean up is necessary if the new option had a note saying "Formerly known as stabilityDays" which would be found when searching the docs.

The maintainer added a note to the `minimumReleaseAge` docs, but also introduced a typo, which I'm fixing with this PR:

- #22527

@TWiStErRob you can put a `X` in your todo about adding a note. 😉

**Reply by TWiStErRob** [reply]
Done, also I see most of the changes are done, thank you very much @rarkins!

@HonkingGoose you could accept the answer here: https://github.com/renovatebot/renovate/discussions/14948

**Reply by HonkingGoose** [reply]
I accepted the answer in that discussion. 🙂

**Reply by HonkingGoose** [reply]
@TWiStErRob I just finished your last two todos, please check my work, and then tick the relevant checkboxes. 😉

**Reply by TWiStErRob** [reply]
Thanks for being open to a bit of admin guys! I really hope this will help other lost souls!

**Reply by HonkingGoose** [reply]
@TWiStErRob You're welcome!
